### ISSUE TITLE
`linktest.c` does not build with clang 15

### ISSUE THREAD

**Issue by eschnett** [xianyi/OpenBLAS#3760]
The file `linktest.c` does not build with Clang 15. The error is
```
linktest.c:2:1: error: call to undeclared function 'lsame_'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
```
Similar errors are reported for all other functions.

I assume that creating declarations such as `void lsame_();` will prevent these errors.

This is with OpenBLAS 0.3.20.

**Comment by eschnett** [comment]
I confirm that this patch resolves the problem for me:
```patch
--- a/exports/gensymbol
+++ b/exports/gensymbol
@@ -3914,6 +3914,18 @@
     @underscore_objs = (@underscore_objs, @misc_common_objs);
     @no_underscore_objs = (@no_underscore_objs, @misc_common_objs);
 
+    foreach $objs (@underscore_objs) {
+        print "void ", $symbolprefix, $objs, $bu, $symbolsuffix, "();\n" if $objs ne "xerbla";
+    }
+
+    foreach $objs (@need_2underscore_objs) {
+        print "void ", $symbolprefix, $objs, $bu, $bu, $symbolsuffix, "();\n";
+    }
+
+    foreach $objs (@no_underscore_objs) {
+        print "void ", $symbolprefix, $objs, $symbolsuffix, "();\n";
+    }
+
     print "int main(void){\n";
     foreach $objs (@underscore_objs) {
         print $symbolprefix, $objs, $bu, $symbolsuffix, "();\n" if $objs ne "xerbla";
```

**Comment by martin-frbg** [comment]
this was fixed post 0.3.21 by #3731, which should apply cleanly to 0.3.20 as well
